pkg/process/checks/allprocesses_linux.go

```diff
@@ -12,15 +12,15 @@ import (
 // getAllProcesses uses a probe to fetch processes using procutil library,
 // then convert them into FilledProcesses for compatibility
 func getAllProcesses(probe *procutil.Probe) (map[int32]*process.FilledProcess, error) {
-	procs, err := probe.ProcessesByPID(time.Now())
+	procs, err := probe.ProcessesByPID(time.Now(), true)
```

Comment:
Why not change this to `ProcessesByPIDWithPerm()`?

Reply:
+1 - lets keep the API as straightforward as possible.

Reply:
Actually I think we should get rid of it completely and have this configuration live in the `Probe{}` (commented below as well)